libs/positions/src/lib/positions-data-providers.ts

```diff
@@ -120,27 +114,26 @@ export const POSITIONS_SUBSCRIPTION = gql`
 `;
 
 export const getMetrics = (
-  data: Positions_party | null,
-  accounts: AccountFieldsFragment[] | null
+  data: PositionRejoined[] | null,
+  accounts: Account[] | null
 ): Position[] => {
-  if (!data || !data?.positionsConnection?.edges) {
+  if (!data || !data?.length) {
     return [];
   }
   const metrics: Position[] = [];
-  data?.positionsConnection.edges.forEach((position) => {
-    const market = position.node.market;
-    const marketData = market.data;
-    const marginLevel = position.node.marginsConnection?.edges?.find(
-      (margin) => margin.node.market.id === market.id
-    )?.node;
+  data.forEach((position) => {
+    const market = position.market;
+    const marketData = market?.data;
+    const marginLevel = position.margins;
     const marginAccount = accounts?.find((account) => {
-      return account.market?.id === market.id;
+      return account.market?.id === market?.id;
     });
     if (
       !marginAccount ||
       !marginLevel ||
+      !market ||
       !marketData ||
-      position.node.openVolume === '0'
+      position.openVolume === '0'
```

Comment:
question (blocking): Why this condition? I think at this point we should show all positions regardless of whether they have volume or not

Reply:
It's basically not my piece of code, and I wouldn't change general flow at this situation. It's rather case for decision of product owner and for separate issue.